src/OutputChecker.cs

```diff
@@ -160,6 +160,11 @@ private string GetOldContent(string file)
             }
         }
 
+        private byte[] GetOldBinaryContent(string file)
+        {
+            return File.ReadAllBytes(file);
```

Comment:
This has to take the staged value from git (same as the `GetOldContent` function does it). Otherwise, on the first run the test will fail, but overwrite the file with the new (probably broken) value. Next run will read the already broken file, and pass. If we read the file from git stage, the user has to explicitly use `git add` to overwrite the file, so just running the tests mutliple times won't fix the test failure.

Reply:
`git cat-file` should just return the binary data to stdout, these low level git commands usually don't care about text/binary distinctions